e2e/tests/core/03-connection/connection_test.go

```diff
@@ -0,0 +1,89 @@
+package connection
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	paramsproposaltypes "github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/test"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	connectiontypes "github.com/cosmos/ibc-go/v5/modules/core/03-connection/types"
+	ibctesting "github.com/cosmos/ibc-go/v5/testing"
+)
+
+func TestConnectionTestSuite(t *testing.T) {
+	suite.Run(t, new(ConnectionTestSuite))
+}
+
+type ConnectionTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+// QueryConnectionEnabledParam queries the on-chain connection enabled param for 03-connection
+func (s *ConnectionTestSuite) QueryMaxExpectedTimePerBlockParam(ctx context.Context, chain ibc.Chain) uint64 {
+	queryClient := s.GetChainGRCPClients(chain).ParamsQueryClient
+	res, err := queryClient.Params(ctx, &paramsproposaltypes.QueryParamsRequest{
+		Subspace: "ibc",
+		Key:      string(connectiontypes.KeyMaxExpectedTimePerBlock),
+	})
+	s.Require().NoError(err)
+
+	// removing additional strings that are used for amino
+	delay := strings.ReplaceAll(res.Param.Value, "\"", "")
+	time, err := strconv.ParseUint(delay, 10, 64)
+	s.Require().NoError(err)
+
+	return time
+}
+
+// TestMaxExpectedTimePerBlock tests changing the MaxExpectedTimePerBlock param using a governance proposal
+func (s *ConnectionTestSuite) TestMaxExpectedTimePerBlock() {
+	t := s.T()
+	ctx := context.TODO()
+
+	_, _ = s.SetupChainsRelayerAndChannel(ctx, transferChannelOptions())
```

Comment:
This step is overkill as the test only requires one chain but the tests fail without the setup.

Reply:
Yea that's one limitation I've noticed with the testing package. It assumes all tests require 2 chains to do the same functionality

Reply:
There's an [issue](https://github.com/cosmos/ibc-go/issues/2012) to make this more flexible with the intention of supporting an environment for multi-hop, I think maybe we could extend that to provide arbitrary chain configs which would make it easy to just create a single chain for tests where that's all we need.

Reply:
if we're ignoring both return values maybe we can just call

```golang
s.SetupChainsRelayerAndChannel(ctx, transferChannelOptions())
```
without explicitly discarding them. I don't have a strong opinion about this, I can see the value in explicitly showing that there is a return value we're not using if that is the intent.

Reply:
We end up using both chains to test that a packet can be received.